frontend/src/components/personal/EmailChangeForm.tsx

```diff
@@ -38,11 +38,18 @@ export default function EmailChangeForm (): React.ReactElement {
     const formik = useFormik({
       enableReinitialize: true,
       initialValues: {
-        email: user? user.email : ''
+        email: user? user.email : '',
+        password: '',
       },
       validationSchema: validationSchema,
       onSubmit: (values) => {
-        console.log('the data is sent', values)
+        if (values.email && !values.password) {
+          handleOpen()
+        }
+        if (values.email && values.password) {
+          console.log('The data is collected', values)
```

Comment:
I think you can add request to API. Even if it won't match in future, it can be easily updated

Reply:
added